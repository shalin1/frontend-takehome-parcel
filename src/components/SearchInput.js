import React from 'react';
import PropTypes from 'prop-types';

const SearchInput = ({onChange,value}) => (
    <div role="search" key="container">
        <label htmlFor="search-input">Search</label>
        <input
            value={value}
            onChange={e=>onChange(e.target.value)}
            placeholder="Search Gems..."
            type="search"
            id="search-input"
            name="search-input"
        />
    </div>
)

SearchInput.propTypes = {
    onChange: PropTypes.func.isRequired,
    value: PropTypes.string.isRequired
}

export default SearchInput