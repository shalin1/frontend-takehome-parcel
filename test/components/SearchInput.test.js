import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import {axe, toHaveNoViolations} from "jest-axe";
expect.extend(toHaveNoViolations);
import SearchInput from '../../src/components/SearchInput';

describe('SearchInput', () => {
    const onChange = jest.fn()
    const props = { value: 'initial value', onChange }
    const wrapper = shallow(<SearchInput {...props} />)
    const input = wrapper.find('input')

    it('renders', () => {
        expect(wrapper.exists()).toBe(true)
    })

    it('has not changed from the last snapshot', () => {
        const tree = renderer.create(wrapper).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('has no programmatically detectable a11y issues', async () => {
        const results = await axe(wrapper.html());
        expect(results).toHaveNoViolations();
    })

    it('renders a label', () => {
        const label = wrapper.find('label')
        expect(label.exists()).toBe(true)
        expect(label.text()).toBe('Search')
    })

    it('renders an input element with the expected placeholder', () => {
        expect(input.exists()).toBe(true);
        expect(input.props().placeholder).toBe('Search Gems...')
    });

    it('sets the value of the input element to the value prop', () => {
        expect(input.props().value).toBe('initial value')
    })

    describe('when the input is updated',()=>{
        it('calls the onChange hook with the updated value', () => {
            const event = {target:{value:'updated value'}}
            input.simulate('change',event);

            expect(onChange).toHaveBeenCalledWith('updated value')
        })
    })
});
