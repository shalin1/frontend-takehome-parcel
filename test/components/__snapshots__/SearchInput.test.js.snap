// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput has not changed from the last snapshot 1`] = `
<div
  role="search"
>
  <label
    htmlFor="search-input"
  >
    Search
  </label>
  <input
    id="search-input"
    name="search-input"
    onChange={[Function]}
    placeholder="Search Gems..."
    type="search"
    value="initial value"
  />
</div>
`;
